src/data/energy.ts

```diff
@@ -401,7 +401,13 @@ const getEnergyData = async (
 
   const dayDifference = differenceInDays(end || new Date(), start);
   const period =
-    dayDifference > 35 ? "month" : dayDifference > 2 ? "day" : "hour";
+    isFirstDayOfMonth(start) &&
+    isLastDayOfMonth(end || new Date()) &&
```

Comment:
```suggestion
    (!end || isLastDayOfMonth(end)) &&
```
If there is no end date we should prefer `month` otherwise we will always use `day` if today is not the LastDayOfMonth

Reply:
makes sense, thanks